File: JavaScriptPractice/Leetcode440.js
var countK=function(prefix,n){
    let curPrefix = prefix
    let nextPrefix = prefix+1
    let ans=0;
    while (curPrefix<=n) {
        ans+=Math.min(n+1,nextPrefix) - curPrefix
        curPrefix*=10
        nextPrefix*=10
    }
    return ans

};
var findKthNumber = function(n, k) {
    let ans=0
    k-=1
    while (k>0) {
        let count = countK(ans,n)
        if (count<=k) {
            k-=count
            ans+=1
        }
        else 
        {
            k-=1
            ans*=10
        }
    }
    return ans
};

File: CSharpPractice/LeetcodePractice/Leetcode440.cs
public class Solution440 {
    private long countK(long prefix,long n)
    {
        long ans=0;
        long curPrefix = prefix;
        long nextPrefix = prefix+1;
        while (curPrefix<=n)
        {
            ans+=Math.Min(n+1,nextPrefix) - curPrefix;
            curPrefix*=10;
            nextPrefix*=10;
        }
        return ans;
    }
    public int FindKthNumber(long n, long k) {
        int prefix = 1;
        k -=1;
        while (k>0)
        {
            long count = countK(prefix,n);
            if (count<=k)
            {
                k-=count;
                prefix+=1;
            }
            else
            {
                k-=1;
                prefix*=10;
            }
        }
        return prefix;
    }
}